sdk/apis/tenancy/v1alpha1/types_workspace.go

```diff
@@ -174,6 +176,23 @@ type WorkspaceSpec struct {
 	//
 	// +kubebuilder:format:uri
 	URL string `json:"URL,omitempty"`
+
+	// Mount is a reference to a mount that is used to mount the workspace.
+	// If specified, logicalcluster will not be created and the workspace will be mounted
+	// using reference mount object.
+	//
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="mount is immutable"
+	Mount *Mount `json:"mount,omitempty"`
+}
+
+func (in *WorkspaceSpec) IsMounted() bool {
+	return in.Mount != nil
+}
+
+type Mount struct {
+	// Reference is an ObjectReference to the object that is mounted.
+	Reference corev1.ObjectReference `json:"ref,omitempty"`
```

Comment:
please use our own variant. This has far too many fields and makes the semantics unclear.

Reply:
done